oxenss/rpc/request_handler.cpp

```diff
@@ -516,6 +516,14 @@ void RequestHandler::process_client_req(rpc::store&& req, std::function<void(Res
             return cb(
                     Response{http::NOT_ACCEPTABLE, "store timestamp too far from current time"sv});
         }
+
+        if (req.subkey) {
+            if (service_node_.get_db().subkey_revoked(*req.subkey)) {
+                log::debug(logcat, "store: subkey previously revoked");
+                return cb(Response{http::UNAUTHORIZED, "store signature verification failed"sv});
+            }
+        }
+
```

Comment:
This same block of code is in three places now (store, retrieve, expire_msgs).  It would be cleaner to move it into a single check in `verify_signature()` (which would then need to also take a reference to the `db` object) so that it automatically applies to anywhere that allows subkey auth.